api/resources_portal/views/document_views.py

```diff
@@ -0,0 +1,330 @@
+        "has_pre_print": "has_pre_print",
+    }
+
+    # Define ordering fields
+    ordering_fields = {
+        "id": "id",
+        "title": "title.raw",
+        "category": "category.raw",
+        "created_at": "created_at",
+        "updated_at": "updated_at",
+    }
+
+    # Specify default ordering
+    ordering = (
+        "_score",
+        "created_at",
+        "id",
+    )
+
+    faceted_search_fields = {
+        "category": {"field": "category", "facet": TermsFacet, "enabled": True},
+        "organism": {"field": "organism", "facet": TermsFacet, "enabled": True},
+        "has_publication": {"field": "has_publication", "facet": TermsFacet, "enabled": True},
+        "has_pre_print": {"field": "has_pre_print", "facet": TermsFacet, "enabled": True},
+    }
+    faceted_search_param = "facet"
+
+    def list(self, request, *args, **kwargs):
+        response = super(MaterialDocumentView, self).list(request, args, kwargs)
+        response.data["facets"] = self.transform_es_facets(response.data["facets"])
+        return response
+
+    def transform_es_facets(self, facets):
+        """Transforms Elastic Search facets into a set of objects where each one corresponds
+        to a filter group. Example:
+
+        { technology: {rna-seq: 254, microarray: 8846, unknown: 0} }
+
+        Which means the users could attach `?technology=rna-seq` to the url and expect 254
+        samples returned in the results.
+        """
+        result = {}
+        for field, facet in iteritems(facets):
+            filter_group = {}
+            for bucket in facet["buckets"]:
+                filter_group[bucket["key"]] = bucket["total_samples"]["value"]
+            result[field] = filter_group
+        return result
+
+
+class OrganizationDocumentSerializer(serializers.Serializer):
+    id = serializers.IntegerField(read_only=True)
+    name = serializers.CharField(read_only=True)
+    owner_id = serializers.CharField(read_only=True)
+    created_at = serializers.DateField(read_only=True)
+    updated_at = serializers.DateField(read_only=True)
+
+    class Meta:
+        model = Organization
+        fields = (
+            "id",
+            "name",
+            "owner_id",
+            "created_at",
+            "updated_at",
+        )
+        read_only_fields = ("id", "created_at", "updated_at")
+
+
+class OrganizationDocumentView(DocumentViewSet):
+    """Defines search and filter fields for Organizations"""
+
+    document = OrganizationDocument
+    serializer_class = OrganizationDocumentSerializer
+    pagination_class = ESLimitOffsetPagination
+
+    # Filter backends provide different functionality we want
+    filter_backends = [
+        FilteringFilterBackend,
+        OrderingFilterBackend,
+        DefaultOrderingFilterBackend,
+        CompoundSearchFilterBackend,
+    ]
+
+    lookup_field = "id"
+
+    # Define search fields
+    # Is this exhaustive enough?
+    search_fields = {"name": {"boost": 10}}
```

Comment:
Can people also search by owner? That seems like it will be the most common use case.

Reply:
Yeah, and also publication/ pre-print authors. But we aren't pulling that information from anywhere yet. I thought we could do that after MVP.

Reply:
Added ability to search by owner and authors.

Reply:
Wait searching materials by authors makes sense. But don't we need to be able to search organizations by their owners?

Reply:
Added ability to search organizations by their owners.